pkg/core/tekton/tekton.go

```diff
@@ -414,6 +414,9 @@ func toTektonTaskSpec(step workflow.WorkflowStep) v1beta1.TaskSpec {
 	for _, stepEnv := range step.Env {
 		tb.Env(*stepEnv.Name, *stepEnv.Value)
 	}
+	// export some useful env variables to all steps
+	tb.Env(envWorkflowNamespace, namespace)
+	tb.Env(envWorkflowName, name)
```

Comment:
I would like to suggest adding a prefix for those common environment variables (e.g. `FUSEML_DEFAULTS_`) to make it difficult for someone to overwrite then when using a env variable with the same name.

Reply:
OK, check the constants now...

Reply:
I am afraid that won't work as `FUSEML_` is already used (see `inputsVarPrefix`).

A better solution might be set a constant with the prefix you choose, have a slice with the environment variables and iterate over it with a switch setting its value and finally adding it to the pipeline with the prefix.

Reply:
> It looks like too complicated solution for such simple case.
> 
> You're basically afraid that when someone has `WORKFLOW_NAMESPACE` variable in the workflow, it will get translated into `FUSEML_WORKFLOW_NAMESPACE` using `inputsVarPrefix`, right?
> 

exactly!

> OK, then when I use the var name as you originally suggested, e.g. `FUSEML_ENV_WORKFLOW_NAMESPACE`, confict avoided.
> 
> Or do you still think I should check the case when user intentionally has `FUSEML_ENV_WORKFLOW_NAMESPACE`? I do not know what the correct solution would be ... probably do the default envs first, letting it possibly be replaced by user defined at the end of the function?


No, I don't think you should check for that, the idea was to make it more easy for adding new variables (just add it to the slice and add a `case` into the `switch`. It's just a suggestion, there might be even a better way to do that in case we have many variables.

Reply:
Ah, you mean the slice with such global env variables should be pre-filled during some init as a global variable too?